conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/UndertowRuntime.java

```diff
@@ -35,4 +35,7 @@
 
     /** Provides the {@link MarkerCallback} to execute arbitrary actions given a marker on a param. */
     MarkerCallback markers();
+
+    /** Provides the {@link AsyncRequestProcessing} used to detach execution from the request thread. */
+    AsyncRequestProcessing async();
```

Comment:
sync implementations don't need the async stuff. possible to separate?

Reply:
the `UndertowRuntime` interface is only meant to be consumed by generated code. Sync implementations will not be generated to invoke `async()`. I think additional separation will make the code unwieldy, but you may have something in mind that I haven't considered.